src/CodeAnalysis/CSharp/CSharpCodeAnalysisDiagnosticAnalyzers.csproj

```diff
@@ -42,6 +42,7 @@
   <ItemGroup>
     <Compile Include="FixAnalyzers\CSharpFixerWithFixAllAnalyzer.cs" />
     <Compile Include="MetaAnalyzers\CSharpDiagnosticDescriptorCreationAnalyzer.cs" />
+    <Compile Include="MetaAnalyzers\CSharpImmutableObjectMethodAnalyzer.cs" />
```

Comment:
+1 on Sri's suggestion. We can put this in root folder for now and move it later into a separate folder for related rules.